src/main.rs

```diff
@@ -164,6 +171,20 @@ fn resolve_dependencies(context: &CliContext) -> Vec<ResolvedDependency> {
         ::std::process::exit(1)
     }
 
+    // If upgrade and there is a lockfile, we want to adjust the resolved dependencies s.t. if the resolved dep has the same
+    // name and version in the lockfile, we say that it was resolved from the lockfile
+    if sync_mode == &SyncMode::FullUpgrade && context.lockfile.is_some() {
+        resolution
+            .found
+            .par_iter_mut()
+            .map(|dep| {
+                if context.lockfile.as_ref().unwrap().contains_resolved_dep(dep) {
+                    dep.from_lockfile = true;
+                }
+            })
+            .collect::<Vec<_>>();
```

Comment:
can we just create a new vec rather than mutating? I also don't think it's going to be super useful for that to be parallel, did you notice a difference?